tools/libfrencutils/mpp_io.c

```diff
@@ -1037,37 +1038,53 @@ void mpp_set_deflation(int fid_in, int fid_out, int deflation, int shuffle) {
         sprintf(errmsg, "mpp_io(mpp_set_deflation): Error in getting determining netcdf version");
         netcdf_error(errmsg, status);
     }
-    printf("Input: filename=%s, nvar=%i, format=%i\n", files[fid_in].name, files[fid_in].nvar, format);
+    //printf("Input: filename=%s, nvar=%i, format=%i\n", files[fid_in].name, files[fid_in].nvar, format);
     if (format == NC_FORMAT_CLASSIC || format == NC_FORMAT_64BIT) {
-        printf("Not compressing because input file is NetCDF3\n");
+        //printf("Not compressing because input file is NetCDF3\n");
         return;
     }
 
-    int v, shuffle2, deflate2, deflation2;
-
-    // loop thru vars
-    for (v = 0; v < files[fid_in].nvar; ++v) {
-        // get existing compression settings
-        status = nc_inq_var_deflate(files[fid_in].ncid, files[fid_in].var[v].fldid, &shuffle2, &deflate2, &deflation2);
-        if (status != NC_NOERR) {
-            sprintf(errmsg, "mpp_io(mpp_set_deflation): Error in getting deflation level");
-            netcdf_error(errmsg, status);
+    // loop thru output vars
+    int vid_in;
+    char varname[VARNAME_MAXSIZE];
+    int shuffle_in, deflate_in, deflation_in;
+    int shuffle_out, deflation_out;
+
+    for (int vid_out = 0; vid_out < files[fid_out].nvar; ++vid_out) {
+        // if user has not specified options, look up and use settings in input file
+        if (user_deflation == -1 || user_shuffle == -1) {
+            mpp_get_varname(fid_in, vid_out, varname);
```

Comment:
Chris, this assumes that the variable_ids are 0,1,2,3,..., (files[fid_out].nvar -1) .  Is this true. I think this is usually true, but is it always true?

Reply:
Thank you! The cargo cult programmer in me thought it would be true or maybe I saw it similar somewhere else and assumed (that's cargo cult also, ha ha!). Thanks for the review- all the in's and out's at least need a strong double-check and then there's things like this that I also took for granted. I agree with you `get_varid` has a similar use..